app/javascript/components/explore/DifferentialExpressionPanel.jsx

```diff
@@ -49,60 +51,233 @@ function DownloadButton({ bucketId, deFilePath }) {
   )
 }
 
+/** A small icon-like button that makes a dot plot */
+function DotPlotButton({ dotPlotGenes, searchGenes }) {
+  const actionColor = '#3D5A87'
+  // Whipped up via https://boxy-svg.com/app,
+  // based on Alexandria-approved mockup at:
+  // https://docs.google.com/presentation/d/1j8zt1Hj4otD593FtkXlBsPw4GsxkU4XOVYXQx3Ec--E/edit#slide=id.g19cbfc5899b_0_9
+  return (
+    <a
+      className="de-dot-plot-button"
+      onClick={() => {searchGenes(dotPlotGenes)}}
+      data-analytics-name="differential-expression-download"
+      data-toggle="tooltip"
+      data-original-title="View dot plot for genes on this DE table page"
+    >
+      <svg viewBox="119.295 104.022 40.338 40.976" width="14" height="14">
+        <ellipse style={{ 'fill': actionColor }} cx="130.295" cy="115.041" rx="11" ry="11"></ellipse>
+        <ellipse style={{ 'fill': actionColor }} cx="153.18" cy="115.779" rx="2.5" ry="2.5"></ellipse>
+        <ellipse style={{ 'fill': actionColor }} cx="128.719" cy="137.129" rx="5" ry="5"></ellipse>
+        <ellipse style={{ 'fill': actionColor }} cx="151.633" cy="136.998" rx="8" ry="8"></ellipse>
+      </svg>
+    </a>
+  )
+}
+
+/**
+ * Icon for current sort order direction in table column header
+ *
+ * @param {String} order Direction of current sort order: 'asc' or 'desc'
+ */
+function SortIcon({ order }) {
+  const isAscending = order === 'asc'
+  const dirIcon = isAscending ? faAngleDown : faAngleUp
+  return (
+    <button className="sort-icon">
+      <FontAwesomeIcon icon={dirIcon}/>
+    </button>
+  )
+}
+
+const columnHelper = createColumnHelper()
+
+/** Search genes from DE table */
```

Comment:
I think "DE" is fine in docstrings, code comments, and variable names -- often even preferable, as we commonly internally write "DE table" instead of "differential expression table", etc.  Writing out "differential expression" (regardless of casing) strikes me as more worthwhile for function names, and especially as a main hook for analytics.  These latter cases often span system boundaries, and thus can't assume as much context.  Docstrings, code comments, and variable names typically have the fuller canonical phrasing right nearby, so abbreviating makes sense to me for economy in reading.

The case for writing out "differential expression" seems stronger for user-facing text, as you note.  I've updated those accordingly.